File: Android软件/back_end/back_end/src/main/java/com/wireless/model/entity/EnvData.java
package com.wireless.model.entity;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.math.BigDecimal;
import java.time.LocalDateTime;

/**
 * 环境数据实体
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class EnvData {
    private Long id;
    private String deviceCode;
    private BigDecimal temperature;
    private BigDecimal humidity;
    private BigDecimal smokeLevel;
    private LocalDateTime createTime;
}

File: Android软件/app/src/main/java/com/smarthome/app/core/model/EnvData.kt
package com.smarthome.app.core.model

import com.squareup.moshi.Json

data class EnvData(
    @Json(name = "id") val id: Long = 0,
    @Json(name = "deviceCode") val deviceCode: String? = null,
    @Json(name = "temperature") val temperature: Double? = null,
    @Json(name = "humidity") val humidity: Double? = null,
    @Json(name = "smokeLevel") val smokeLevel: Double? = null,
    @Json(name = "createTime") val createTime: String? = null
)

data class EnvChartVO(
    @Json(name = "times") val times: List<String> = emptyList(),
    @Json(name = "temperatures") val temperatures: List<Double> = emptyList(),
    @Json(name = "humidities") val humidities: List<Double> = emptyList(),
    @Json(name = "smokeLevels") val smokeLevels: List<Double> = emptyList(),
    @Json(name = "summary") val summary: String? = null
)
